client/configuration_variable.go

```diff
@@ -120,15 +110,15 @@ func (client *ApiClient) ConfigurationVariablesByScope(scope Scope, scopeId stri
 		return []ConfigurationVariable{}, err
 	}
 
-	// The API returns global and template scopes for environment (and other) scopes. Filter them out.
-	if scope != ScopeGlobal {
-		result = filterOutConfigurationVariables(result, ScopeGlobal)
-		if scope != ScopeTemplate {
-			result = filterOutConfigurationVariables(result, ScopeTemplate)
+	// The API returns variables of upper scopes. Filter them out.
+	var filteredVariables []ConfigurationVariable
+	for _, variable := range result {
+		if scope == variable.Scope {
+			filteredVariables = append(filteredVariables, variable)
 		}
```

Comment:
that's not good enough 😅 
as per the explanation in the issue https://github.com/env0/terraform-provider-env0/issues/751

There'd be several variables from the same scope "type" from a different specific scope.
you'd get more variables from the API in the `PROJECT` scope, but which are NOT defined on the requested `projectId`.

Does that make sense?

Reply:
Each variable has a `scope` and a `scopeId`.
The scope can be `ENVIRONMENT`, `PROJECT`, etc.
The scopeIds are uuids fir the specific environment, project, etc. respectfully.

Reply:
a

Reply:
so.. should I filter instead based on scope ID? (instead of scope).

Reply:
YES

Reply:
updated.